packages/eas-cli/src/credentials/ios/actions/SetUpSubmissionCredentials.ts

```diff
@@ -0,0 +1,68 @@
+import chalk from 'chalk';
+import nullthrows from 'nullthrows';
+import wrapAnsi from 'wrap-ansi';
+
+import { CommonIosAppCredentialsFragment } from '../../../graphql/generated';
+import Log, { learnMore } from '../../../log';
+import { promptAsync } from '../../../prompts';
+import { CredentialsContext } from '../../context';
+import { UnsupportedCredentialsChoiceError } from '../../errors';
+import { AppLookupParams } from '../api/GraphqlClient';
+import { AppStoreApiKeyPurpose } from './AscApiKeyUtils';
+import { SetUpAscApiKey } from './SetUpAscApiKey';
+
+export const PROMPT_FOR_APP_SPECIFIC_PASSWORD = 'PROMPT_FOR_APP_SPECIFIC_PASSWORD';
+export class SetUpSubmissionCredentials {
+  private setupAscApiKeyAction;
+  constructor(app: AppLookupParams) {
+    this.setupAscApiKeyAction = new SetUpAscApiKey(app, AppStoreApiKeyPurpose.SUBMISSION_SERVICE);
+    this.setupAscApiKeyAction.choices = this.setupAscApiKeyAction.choices.concat({
```

Comment:
I know why you did it but modifying object members is a bad anti-pattern. Add a comment explaining why we need this.